### PR TITLE
Add test cases for ORC writing according to options orc.compress and compression [databricks]

### PR DESCRIPTION
**Pull request by res-life**
closes [#8781](https://github.com/NVIDIA/spark-rapids/issues/8781)
closes [#8782 ](https://github.com/NVIDIA/spark-rapids/issues/8782)

ORC writing first honor "compression" option, then "orc.compress" option.

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
321cdh and 330cdh compile failed. Will fix later.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Review by revans2 — DISMISSED** [review]

**Review by jlowe — COMMENTED** [review]
Needs to be upmerged to solve merge conflicts but otherwise lgtm.

File: tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala [hunk 1]
```diff
@@ -92,4 +102,76 @@ class OrcQuerySuite extends SparkQueryCompareTestSuite {
   ) {
     frame => frame
   }
+
+  private def getOrcFilePostfix(compression: String): String =
```
Comment (mythrocks) [inline tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala:106]:
```suggestion
  private def getOrcFileSuffix(compression: String): String =
```

File: tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala [hunk 2]
```diff
@@ -92,4 +102,76 @@ class OrcQuerySuite extends SparkQueryCompareTestSuite {
...
+        assert(maybeOrcFile.isDefined)
+
+        // check the compress type using ORC jar
+        val orcFilePath = new Path(maybeOrcFile.get.getAbsolutePath)
+        val conf = OrcFile.readerOptions(new Configuration())
+
+        // the reader is not a AutoCloseable for Spark CDH, so use `withResourceIfAllowed`
+        // 321cdh uses lower ORC: orc-core-1.5.1.7.1.7.1000-141.jar
+        // 330cdh uses lower ORC: orc-core-1.5.1.7.1.8.0-801.jar
+        withResourceIfAllowed(OrcFile.createReader(orcFilePath, conf)) { reader =>
```
Comment (mythrocks) [inline tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala:137]: This is a useful utility. It makes the code quite streamlined.

File: tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala [hunk 3]
```diff
@@ -92,4 +102,76 @@ class OrcQuerySuite extends SparkQueryCompareTestSuite {
...
+    supportedWriteCompressType
+  }
+
+  test("SPARK-16610: Respect orc.compress (i.e., OrcConf.COMPRESS) when compression is unset") {
+    // Respect `orc.compress` (i.e., OrcConf.COMPRESS).
+    supportedWriteCompressTypes.foreach { orcCompress =>
+      checkCompressType(None, Some(orcCompress))
+    }
+
+    // make paris, e.g.: [("UNCOMPRESSED", "NONE"), ("NONE", "SNAPPY"), ("SNAPPY", "ZSTD") ... ]
```
Comment (mythrocks) [inline tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala:163]:
```suggestion
    // make pairs, e.g.: [("UNCOMPRESSED", "NONE"), ("NONE", "SNAPPY"), ("SNAPPY", "ZSTD") ... ]
```

**Review by mythrocks — DISMISSED** [review]
Good stuff. Some minor nitpicks, but it looks pretty good to me.

**Comment by res-life** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**